src/components/common/StaffComments.vue

```diff
@@ -0,0 +1,252 @@
+<template>
+  <v-menu
+    bottom
+    nudge-top="33"
+    nudge-left="20"
+    v-model="showComments"
+    :close-on-click="false"
+    :close-on-content-click="false"
+  >
+    <!-- the button -->
+    <template v-slot:activator="{ on, attrs }">
+      <v-btn
+        small text color="primary"
+        id="comments-button"
+        v-bind="attrs"
+        v-on="on"
+      >
+        <v-icon medium>mdi-comment-text-outline</v-icon>
+        <span>{{numComments}}</span>
+      </v-btn>
+    </template>
+
+    <!-- the menu (dialog) -->
+    <v-card flat id="staff-comment-container" class="px-8 py-6">
+      <v-card-title class="d-flex justify-space-between pa-0">
+        <div>
+          <v-icon medium color="primary">mdi-comment-text-outline</v-icon>
+          <span>{{numComments}}</span>
+        </div>
+        <v-btn icon large class="mr-n3" @click="close()">
+          <v-icon color="primary">mdi-close</v-icon>
+        </v-btn>
+      </v-card-title>
+
+      <v-card-text class="mt-2 pa-0">
+        <v-textarea
+          ref="textarea"
+          autofocus
+          no-resize
+          filled
+          rows="5"
+          v-model="comment"
+          placeholder="Enter Comments"
+          :rules="rules"
+        />
+      </v-card-text>
+
+      <v-card-actions class="d-flex justify-space-between pa-0">
+        <div class="body-2 mt-1">{{charsRemaining}}</div>
+        <div class="mr-n3">
+          <v-btn text color="primary" class="font-weight-bold" :loading="isSaving" @click="save()">
+            Save
+          </v-btn>
+          <v-btn text color="primary" @click="close()">
+            Cancel
+          </v-btn>
+        </div>
+      </v-card-actions>
+
+      <v-card-text class="mt-6 pa-0">
+        <div class="existing-comments pr-5">
+          <div v-for="(comment, i) in comments" :key="i" class="body-2">
+            <p class="pre-line" v-html="comment.comment" />
+            <p class="font-italic">
+              {{ comment.submitterDisplayName }}
+              &hyphen;
+              {{apiToSimpleDateTime(comment.timestamp)}}
+            </p>
+          </div>
+        </div>
+      </v-card-text>
+    </v-card>
+  </v-menu>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop } from 'vue-property-decorator'
+import { CommentIF, FormIF } from '@/interfaces'
+import { DateMixin, FilingMixin } from '@/mixins'
+
+@Component({})
+export default class StaffComments extends Mixins(DateMixin, FilingMixin) {
```

Comment:
DateMixin is needed for `apiToSimpleDateTime`.

FilingMixin is needed for `flattenAndSortComments`.

Both of these will need to be added to bcrs-shared-component framework if they're not already there.